### PR TITLE
Add new video components and new icon for play button

### PR DESCRIPTION
**Pull request by lmounsey**
# Description

[Video - Landscape](https://deploy-preview-1599--pattern-lib-unimelb.netlify.app/iframe.html?selectedKind=Embed&selectedStory=Video%20%28Today%29%20-%20Landscape)
[Video - Portrait](https://deploy-preview-1599--pattern-lib-unimelb.netlify.app/iframe.html?selectedKind=Embed&selectedStory=Video%20%28Today%29%20-%20Portrait)
[Video - Full Width](https://deploy-preview-1599--pattern-lib-unimelb.netlify.app/iframe.html?selectedKind=Embed&selectedStory=Video%20%28Today%29%20-%20Full%20width)

New components added - `Video` and `VideoFullWidth`. Have documented these in Storybook and detailed how to use them.

I've also added another icon `play-circle` as well. 

Happy to take your guidance on naming convention of the component here - wasn't sure if should sit with the embed stuff or be seperate.

Also, I don't have any good example assets yet of the 'Portrait' video with an autoloop, so it's still referring to then Brown assets - if this is an issue, do you have any placeholder assets I could be pointing to?

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

# Checklist:

- [x] Make sure to do not repeat yourself (DRY)
- [x] Snapshots tested
- [x] A11y tested
- [x] CSS utilises BEM naming convention and structure
- [x] Code does not mutate variables/objects/arrays but instead uses `map`, `filter` etc to return a new array/object
- [x] Crossbrowser testing (IE11, Safari 8+, iOS 8.4+, Android 4.4+, Firefox ESR (v52.x), iPhone (4s,6), iPad 2, Galaxy s5) 
- [x] My changes generate no new warnings
- [x] I have added tests (e2e and unit) that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Added the new component to the index.js (Vue and Lib) export

# Reviewer Checklist

- [ ] Code is not repeated (DRY)
- [ ] ES6+ code is used where possible
- [ ] Code does not mutate variables/objects/arrays but instead uses `map`, `filter` etc to return a new array/object
- [ ] CSS utilises BEM naming convention and structure
- [ ] Snapshots tested against a copy of dev snapshots to check UI changes are intended
- [x] A11y tested
- [ ] Crossbrowser tested in at least IE11

### CODE REVIEW

**Comment by gepifanio** [comment]
Comment: Deploy Started

Building with commit fc9c9ff71b152180f5325aab3d6b650b7a7f774f

https://app.netlify.com/sites/pattern-lib-unimelb/deploys/6090b48e5806d500083907ce

**Review by jaredh-um — APPROVED** [review]
Hi Lucas,

Looks good to me. I will leave it to Gui regarding the naming conventions and location. We should definitely remove the external asset links as soon as we have our own example though.

Edit: Added alternative URLs for those assets

File: components/video/stories/video.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+- `autoplay`: Boolean for autoplaying the looping video if one exists, otherwise only loops on mouseover
+- `videoApi`: Boolean to add additional query params for analytics
+
+### Example:
+
+```html
+<video
+  video-api
+  video="{ url: 'https://www.youtube.com/embed/sWHoKq6MSaE', duration: 144 }"
+  poster="{ src: 'https://picsum.photos/seed/1/540/360', alt: ' }"
```
Comment (gepifanio) [inline components/video/stories/video.md:18]: alt prop has a typo, is missing one more `'`

File: targets/lib/index.js [hunk 2]
```diff
@@ -232,6 +233,8 @@ Vue.component('campaign-banner', CampaignBanner);
 Vue.component('card-image-link', CardImageLink);
 Vue.component('cards-filter-category', CardsFilterCategory);
 Vue.component('courses-filter', CoursesFilter);
+Vue.component('video', Video);
```
Comment (gepifanio) [inline targets/lib/index.js:236]: good one also needed to be added to the vue build process as well: `targets/vue/index.js`

**Review by gepifanio — CHANGES_REQUESTED** [review]

**Comment by gepifanio** [comment]
> Hi Lucas,
> 
> Looks good to me. I will leave it to Gui regarding the naming conventions and location. We should definitely remove the Brown link as soon as we have our own example though.

Regarding the stories folder, would be better to keep consistent and add the stories under the `Embed` folder.

File: components/video/stories/VideoPortrait.vue [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<template>
+  <Video
+    :video="{
+      url: 'https://www.youtube.com/embed/MH5pJ4jokpI',
+      duration: 28,
+      preview: 'https://player.vimeo.com/external/336863136.sd.mp4?s=234cbb2adf6f8c523093bc0627e7d8bdb697db94&amp;profile_id=165',
+    }"
+    :poster="{
+      src: 'https://www.brown.edu/sites/g/files/dprerj316/files/styles/portrait_sfts_desktop/public/Byrd_Advice.jpg?h=30f5acd3&itok=hcyHpFiN',
```
Comment (jaredh-um) [inline components/video/stories/VideoPortrait.vue:9]: Please replace this URL with the temporary placeholder: 
`https://study.unimelb.edu.au/__data/assets/image/0021/265404/Byrd_Advice.jpg`

File: components/video/stories/VideoPortrait.vue [hunk 4]
```diff
@@ -0,0 +1,27 @@
+<template>
+  <Video
+    :video="{
+      url: 'https://www.youtube.com/embed/MH5pJ4jokpI',
+      duration: 28,
+      preview: 'https://player.vimeo.com/external/336863136.sd.mp4?s=234cbb2adf6f8c523093bc0627e7d8bdb697db94&amp;profile_id=165',
```
Comment (jaredh-um) [inline components/video/stories/VideoPortrait.vue:6]: Please replace this URL with the temporary placeholder:
`https://study.unimelb.edu.au/__data/assets/file/0022/265405/Byrd_Video.mp4`

File: components/embed/Video.vue [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+      const seconds = secNum - (hours * 3600) - (minutes * 60); // eslint-disable-line no-magic-numbers
+
+      return (hours ? `${hours}h ` : '') + (minutes ? `${minutes}m ` : '') + (seconds ? `${seconds}s` : '');
+    },
+    videoSrc() {
+      if (!this.video || !this.video.url) {
+        return '';
+      }
+
+      const additionalParams = this.videoApi ? `&enablejsapi=1&origin=https%3A%2F%2F${document.location.hostname}.unimelb.edu.au` : '';
```
Comment (gepifanio) [inline components/embed/Video.vue:143]: - [ ] Please remove the `unimelb.edu.au` as the `hostname` will already have the unimelb domain and that will include any subdomain as well.

**Review by gepifanio — CHANGES_REQUESTED** [review]

File: components/embed/Video.vue (hunk 5 above)
Comment (lmounsey) [reply components/embed/Video.vue]: This should be fixed now

**Review by gepifanio — APPROVED** [review]
Looks good

**Review by jaredh-um — APPROVED** [review]
Looks good 👍

**Review by gepifanio — CHANGES_REQUESTED** [review]
I forgot to ask and mention before, but could you please also add unit testing for the component, you could follow the same example like this one I am doing for the PathfinderToday:

```

import {
  shallow,
} from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import PathfinderToday from '../PathfinderToday.vue';

expect.extend(toHaveNoViolations);

const pathfinderLinks = [
  {
    title: 'Undergraduate courses',
    description: 'Flexible first degrees that you can tailor to your career goals, or explore new possibilities.',
    href: '/landing-page/',
  },
  {
    title: 'Graduate courses',
    description: 'Study in small classes led by experts alongside others who share your interests and goals.',
    href: '/landing-page/',
  },
  {
    title: 'Graduate research',
    description: 'Access dedicated research centres, advanced  facilities, generous scholarships and more.',
    href: '/landing-page/',
  },
  {
    title: 'Short&nbsp;courses&nbsp;and microcredentials',
    description: 'Upskill, expand your knowledge and take the next step in your career with focused courses.',
    href: '/landing-page/',
  },
];

describe('PathfinderToday', () => {
  it('should match snapshot', () => {
    const result = shallow(PathfinderToday, {
      propsData: {
        items: pathfinderLinks,
      },
    }).element;
    expect(result).toMatchSnapshot();
  });

  it('Component throws no accessibility violations', (done) => {
    const html = shallow(PathfinderToday, {
      propsData: {
        items: pathfinderLinks,
      },
    }).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
```

**Review by gepifanio — APPROVED** [review]

**Comment by jaredh-um** [comment]
A few more refinements:
- instead of just pausing on `mouseout`, should also return video to 0 (start)
- when autoplay finishes for the first time, should also return to 0 and pause
- should only autoplay when in viewport - eg; needs to be fixed in Firefox

**Review by anormand — CHANGES_REQUESTED** [review]
1. Button focus

Problem
- After pressing the close button, focus stays on the close button

Remedy
- After pressing the close button, move focus back to the play button

2. Storybook documentation

We need to included the expected screen reader announcement in the Storybook documentation (see below).

<h4>Screen reader announcement (Voiceover)</h4>

Play button
- Play video [Video label 2m 24s], button

Close button
- Close video, button

**Comment by anormand** [comment]
Video component now works, but Storybook documentation still needs to be updated

**Comment by anormand** [comment]
Now passes accessibility checks

**Pull request merged**